### PR TITLE
fix(ext/node): add missing `release` property to node's `process`

### LINKED ISSUE #18870: npm incompatibility due to process.release.name
## Description

When loading a CommonJS module from npm in deno I'm finding that `process.release.name` is not defined on the process object.

A minimal reproduction that I think represents the same issue is creating a `main.mjs` with contents:
```js
import process from 'node:process';
console.log(`process release name: ${process.release?.name}`);
```

And running it via node vs deno:
```
C:\>node main.mjs     
process release name: node

C:\>deno run main.mjs
process release name: undefined

C:\>deno --version
deno 1.32.5 (release, x86_64-pc-windows-msvc)
v8 11.2.214.9
typescript 5.0.3

C:\>node --version
v18.16.0
```

The [`process.release`](https://nodejs.org/api/process.html#processrelease) docs.

## Context

Trying to load the pyodide npm package as a CommonJS module. I believe when this issue is fixed loading pyodide could be validated with:

```js
import {loadPyodide} from "npm:pyodide/pyodide.js";

const pyodide = await loadPyodide();
await pyodide.loadPackage('micropip');
const micropip = pyodide.pyimport("micropip");
await micropip.install('snowballstemmer');
const result = await pyodide.runPythonAsync(`
import snowballstemmer
stemmer = snowballstemmer.stemmer('english')
stemmer.stemWords('go goes going gone'.split())
`);

console.log(result.toString());
```

Resulting in:

```
C:\>deno run --node-modules-dir -A main.ts
Loading micropip, packaging
Loaded packaging, micropip
['go', 'goe', 'go', 'gone']
```

The issue is that a [heuristic for pyodide to check for execution in node](https://github.com/pyodide/pyodide/blob/24957acc139208a0a8113d185ae55dc9288f75f9/src/js/compat.ts#L4-L9) uses `process.release.name` and is failing due to this issue.

### PR DESCRIPTION
**Pull request by yardenshoham**
Docs: https://nodejs.org/api/process.html#processrelease

```ts
import process from 'node:process';
console.log(process.release);
// outputs:
// {
//   name: "node",
//   sourceUrl: "https://nodejs.org/download/release/v18.12.1/node-v18.12.1.tar.gz",
//   headersUrl: "https://nodejs.org/download/release/v18.12.1/node-v18.12.1-headers.tar.gz"
// }
```

- Fixes #18870

### CODE REVIEW

**Comment by yardenshoham** [comment]
I made it use `deno` for the name and not `node`. Also, I used Deno's source URL instead of node's, is that fine?

File: ext/node/polyfills/process.ts [hunk 1]
```diff
@@ -331,6 +331,16 @@ class Process extends EventEmitter {
     super();
   }
 
+  /** https://nodejs.org/api/process.html#processrelease */
+  get release() {
+    return {
+      name: "deno",
+      sourceUrl: `https://github.com/denoland/deno/archive/refs/tags/v${Deno.version.deno}.tar.gz`,
```
Comment (bartlomieju) [inline ext/node/polyfills/process.ts:338]: Use `versions.deno` intead

Comment (yardenshoham) [reply ext/node/polyfills/process.ts]: Done

**Comment by yardenshoham** [comment]
I wonder if we should set `name` to `node` to be fully compatible with node. But that feels like lying

**Comment by rajsite** [comment]
> I wonder if we should set name to node to be fully compatible with node. But that feels like lying

@yardenshoham if it was changed to 'deno' that wouldn't address the original issue I filed which was a library using the process.release === 'node' to detect environment (browser vs worker vs node). The `node` check would still fail.

Guess it depends on deno's intent with the `npm:` namespace. If the intention is strictly to be a compatibility layer then I'd expect `process.release' to report `node`. Seems like Deno's intent is exisiting npm libraries can be used in Deno without updates. Not to make the `npm:` namespace a first class API target, i.e. the browser API's are Deno's first class APIs that users should be encouraged to target, so would only focus `npm:` on compatibility.

If I was a library author trying to support Deno and Node and needed conditional logic / environment specific behaviors then I'd be checking for the `Deno` global existence. But everything under the `npm:` namespace would be trying to emulate the node environment as closely as possible. i.e. don't want to encourage deno authors to be writing against the `npm:` namespace so Deno wouldn't expose deno-specific behavior there.

**Comment by rajsite** [comment]
> I wonder if we should set `name` to `node` to be fully compatible with node. But that feels like lying

Or to put it another way. I think for `node:` and `npm:` scoped packages Deno actively wants to lie and make it look like a normal node environment to the modules.

**Comment by yardenshoham** [comment]
I agree, I changed it

**Comment by kt3k** [comment]
`name: node` makes sense to me as its for compatibility.

@yardenshoham Can you run `./tools/format.js` to fix format issue in CI?

**Comment by yardenshoham** [comment]
I think it failed on a flaky test

**Review by lino-levan — APPROVED** [review]
LGTM

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**